internal/controllers/bundledeployment/interfaces.go

```diff
@@ -7,15 +7,28 @@ import (
 	"helm.sh/helm/v3/pkg/chart"
 	"helm.sh/helm/v3/pkg/chartutil"
 
-	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+	rukpakv1alpha2 "github.com/operator-framework/rukpak/api/v1alpha2"
 )
 
 type Handler interface {
-	Handle(context.Context, fs.FS, *rukpakv1alpha1.BundleDeployment) (*chart.Chart, chartutil.Values, error)
+	Handle(context.Context, fs.FS, *rukpakv1alpha2.BundleDeployment) (*chart.Chart, chartutil.Values, error)
 }
 
-type HandlerFunc func(context.Context, fs.FS, *rukpakv1alpha1.BundleDeployment) (*chart.Chart, chartutil.Values, error)
+type HandlerFunc func(context.Context, fs.FS, *rukpakv1alpha2.BundleDeployment) (*chart.Chart, chartutil.Values, error)
 
-func (f HandlerFunc) Handle(ctx context.Context, fsys fs.FS, bd *rukpakv1alpha1.BundleDeployment) (*chart.Chart, chartutil.Values, error) {
+func (f HandlerFunc) Handle(ctx context.Context, fsys fs.FS, bd *rukpakv1alpha2.BundleDeployment) (*chart.Chart, chartutil.Values, error) {
 	return f(ctx, fsys, bd)
 }
+
+// TODO: Having two interfaces with same parameters seems unnecessary. This should ideally
+// be moved to HandleBundleDeployment. With it, we can remove the additional step of loading from
+// store.
```

Comment:
I think it probably makes sense to keep storing the unpacked contents. That way future reconciles can see that the data is already present and skip the unpack step.

But I agree that this interface is unnecessary. I think we can just store whatever comes from the source unmodified, and then any templating or modification could happen at the beginning of the Handler function.

Reply:
Modified in https://github.com/operator-framework/rukpak/commit/fd262e028de9b2bb81082408b1a7c12406dcaa56